itools-packager/src/main/java/com/powsybl/itools/ItoolsPackagerMojo.java

```diff
@@ -79,33 +81,125 @@ public void setFiles(File[] files) {
-                    return FileVisitResult.CONTINUE;
+    public void createPackage(Path dir, final Path baseDir, String fileName, String packageName) throws IOException {
+        Objects.requireNonNull(dir);
+        Objects.requireNonNull(baseDir);
+        Objects.requireNonNull(fileName);
+
+        String extension = compareFileExtension(fileName);
+        String newFileName = fileName;
+        // invalid extension
+        if (extension.equals("")) {
+            extension = ".zip";
+            newFileName = packageName + extension;
+        }
+
+        Path packageFilePath = baseDir.resolve(newFileName);
+        FileOutputStream fos = new FileOutputStream(String.valueOf(packageFilePath));
+
+        getLog().info("Generate package file : " + newFileName);
+
+        switch (extension) {
+            case ".zip":
+                try (ZipArchiveOutputStream zos = new ZipArchiveOutputStream(fos)) {
+                    Files.walkFileTree(dir, new SimpleFileVisitor<Path>() {
+                        @Override
+                        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                            ZipArchiveEntry entry = new ZipArchiveEntry(baseDir.relativize(file).toString());
+                            if (Files.isExecutable(file)) {
+                                entry.setUnixMode(0100770);
+                            } else {
+                                entry.setUnixMode(0100660);
+                            }
+                            zos.putArchiveEntry(entry);
+                            Files.copy(file, zos);
+                            zos.closeArchiveEntry();
+                            return FileVisitResult.CONTINUE;
+                        }
+
+                        @Override
+                        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+                            zos.putArchiveEntry(new ZipArchiveEntry(baseDir.relativize(dir).toString() + "/"));
+                            zos.closeArchiveEntry();
+                            return FileVisitResult.CONTINUE;
+                        }
+                    });
+                }
+                break;
+
+            case ".tar.gz":
+                try (TarArchiveOutputStream tos = new TarArchiveOutputStream(new GZIPOutputStream(new BufferedOutputStream(fos)))) {
+                    tos.setBigNumberMode(TarArchiveOutputStream.BIGNUMBER_STAR); // to store big number in the archive.
+                    tos.setLongFileMode(TarArchiveOutputStream.LONGFILE_GNU); // to store long file names in the archive.
+
+                    Files.walkFileTree(dir, new SimpleFileVisitor<Path>() {
+                        @Override
+                        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                            TarArchiveEntry entry = new TarArchiveEntry(new File(file + "/"));
```

Comment:
In case of a file, you should not concat the file with a "/".

Reply:
"/" already exists in zip code. I replace it with File.separator.